Components/Queues/PhpAmqpLibQueueManager.php

```diff
@@ -0,0 +1,309 @@
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPQueueException
+     */
+    public function declareQueue(string $name, int $flag = AMQP_DURABLE, array $arguments = [])
+    {
+        if (!$this->channel) {
+            throw new \AMQPChannelException('There is no channel available');
+        }
+
+        $this->queue = $name;
+        $durable = $flag === AMQP_DURABLE ? true : false;
+        return $this->channel->queue_declare($this->queue, false, $durable, false, false, false, new AMQPTable([$arguments]));
+    }
+
+    /**
+     * Publish a message
+     * @param AMQPMessage $message
+     */
+    public function publish(AMQPMessage $message)
+    {
+        $this->channel->basic_publish($message, self::EXCHANGE_NAME, $this->queue);
+    }
+
+    /**
+     * Add a connection to the browker
+     * @param $connection
+     * @return $this
+     */
+    public function addConnection($connection): self
+    {
+        $this->connections[] = $connection;
+
+        return $this;
+    }
+
+    /**
+     * Returns all the connections available
+     * @return \AMQPConnection[]|array|null
+     */
+    public function getConnections():? array
+    {
+        return $this->connections;
+    }
+
+    /**
+     * Catch a channel inside the connection
+     * @return AMQPChannel
+     */
+    public function declareChannel(): AMQPChannel
+    {
+        $this->channel = reset($this->connections)->channel();
+        $this->channel->basic_qos(null, self::PREFETCH_COUNT, null);
+        return $this->channel;
+    }
+
+    /**
+     * AMQP Exchange is the publishing mechanism
+     */
+    public function declareExchange($exchange): void
+    {
+        $this->exchange = $exchange;
+        $this->exchange->setName(self::EXCHANGE_NAME);
+        $this->channel->exchange_declare($this->exchange->getName(), AMQPExchangeType::DIRECT, false, true, false);
+        $this->channel->queue_bind($this->queue, $this->exchange->getName(), $this->queue);
+    }
+
+    /**
+     * Return the Queue name if there is already a queue. Otherwise it declares a new queue
+     *
+     * @param string $queueName
+     * @return \AMQPQueue
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPException
+     * @throws \AMQPQueueException
+     */
+    public function getQueue(string $queueName): \AMQPQueue
+    {
+        if (!$this->queue) {
+            $this->connect();
+            $this->declareQueue($queueName);
+        }
+        return $this->queue ;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getId():? int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param mixed $id
+     * @return PhpAmqpLibQueueManager
+     */
+    public function setId($id): self
+    {
+        $this->id = $id;
+        return $this;
+    }
+
+    /**
+     * Shutdown the connection
+     * @throws \Exception
+     */
+    public function shutdown(): bool
```

Comment:
Which is the difference between shutdown and disconnect?

Reply:
shutdown was removed